daemon/cluster/convert/service.go

```diff
@@ -56,24 +64,42 @@ func ServiceFromGRPC(s swarmapi.Service) types.Service {
 		service.UpdateStatus.Message = s.UpdateStatus.Message
 	}
 
-	return service
+	return service, nil
 }
 
-func serviceSpecFromGRPC(spec *swarmapi.ServiceSpec) *types.ServiceSpec {
+func serviceSpecFromGRPC(spec *swarmapi.ServiceSpec) (*types.ServiceSpec, error) {
 	if spec == nil {
-		return nil
+		return nil, nil
 	}
 
 	serviceNetworks := make([]types.NetworkAttachmentConfig, 0, len(spec.Networks))
 	for _, n := range spec.Networks {
 		serviceNetworks = append(serviceNetworks, types.NetworkAttachmentConfig{Target: n.Target, Aliases: n.Aliases})
 	}
 
+	taskTemplate := taskSpecFromGRPC(spec.Task)
+
+	switch t := spec.Task.Runtime.(type) {
```

Comment:
Perhaps you could do a `switch runtime := naming.Runtime(spec.Task)` here and switch on the string (`container` vs `com.docker.runtime.plugin`. @aaronlehmann?

If we keep it this way, then rather than switching on `switch t.Generic.Payload.TypeUrl {` we should switch on `t.Generic.Kind`